From 1ca6b35a500348620c98696ac8e7a344c0bce165 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 14 May 2013 10:45:07 -0500
Subject: [PATCH 153/210] Fix builds on PowerBook G4

On the PowerBook G4, a 'uname -m' command returns "ppc"; however,
the architecture-specific files are located at arch/powerpc/... The
Makefile is modified to transform the ARCH variable appropriately,
and the appropriate little/big-endian mode is set. In addition, the
code in rtw_br_ext.c skips the call to the missing routine csum_ipv6_magic().
The latter change is temporary.

Although the driver can be built on the PowerPC architecture, it does not
yet work on that architecture.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 Makefile          | 8 ++++++--
 core/rtw_br_ext.c | 2 ++
 2 files changed, 8 insertions(+), 2 deletions(-)

--- a/drivers/staging/rtl8192du/Makefile
+++ b/drivers/staging/rtl8192du/Makefile
@@ -115,9 +115,13 @@ ifeq ($(CONFIG_INTEL_WIDI), y)
 EXTRA_CFLAGS += -DCONFIG_INTEL_WIDI
 endif
 
-EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
-SUBARCH := $(shell uname -m | sed -e s/i.86/i386/)
+SUBARCH := $(shell uname -m | sed -e s/i.86/i386/ | sed -e s/ppc/powerpc/)
 ARCH ?= $(SUBARCH)
+ifeq ($(ARCH), "powerpc")
+EXTRA_CFLAGS += -DCONFIG_BIG_ENDIAN
+else
+EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
+endif
 CROSS_COMPILE ?=
 KVER  := $(shell uname -r)
 KSRC := /lib/modules/$(KVER)/build
--- a/drivers/staging/rtl8192du/core/rtw_br_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_br_ext.c
@@ -1266,10 +1266,12 @@ int nat25_db_handle(struct rtw_adapter *
 								      GET_MY_HWADDR(priv))) {
 						struct icmp6hdr  *hdr = (struct icmp6hdr *)(skb->data + ETH_HLEN + sizeof(*iph));
 						hdr->icmp6_cksum = 0;
+#ifdef csum_ipv6_magic
 						hdr->icmp6_cksum = csum_ipv6_magic(&iph->saddr, &iph->daddr,
 										iph->payload_len,
 										IPPROTO_ICMPV6,
 										csum_partial((__u8 *)hdr, iph->payload_len, 0));
+#endif
 					}
 				}
 			}
